app-header-layout/app-header-layout.html

```diff
@@ -0,0 +1,182 @@
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../../polymer/polymer.html">
+<link rel="import" href="../../iron-resizable-behavior/iron-resizable-behavior.html">
+
+<!--
+`app-header-layout` is a layout element that contains a header and a content.
+This layout element uses the document scroll by default, but this element can 
+also define its own scrolling region.
+
+Example:
+
+Using the document scroll:
+
+    <app-header-layout>
+      <app-header condenses waterfall>
+        <app-toolbar>
+          <div title>App name</div>
+        </app-toolbar>
+      </app-header>
+      <div>
+        <!\-\- content goes here \-\->
+      </div>
+    </app-header-layout>
+
+Using an own scrolling region:
+
+    <app-header-layout has-scrolling-region class="fit">
+      <app-header condenses waterfall>
+        <app-toolbar>
+          <div title>App name</div>
+        </app-toolbar>
+      </app-header>
+      <div>
+        <!\-\- content goes here \-\->
+      </div>
+    </app-header-layout>
+
+@demo demo1.html
+@demo demo2.html
+-->
+
+<dom-module id="app-header-layout">
+
+  <style>
+
+    :host {
+      display: block;
+    }
+
+    :host([has-scrolling-region]) {
+      position: relative;
+      height: 100%;
+    }
+
+    :host > ::content > app-header {
+      position: fixed;
+      top: 0;
+      right: 0;
+      left: 0;
+      z-index: 2;
+    }
+
+    :host([has-scrolling-region]) > ::content > app-header {
+      position: absolute;
+    }
+
+    :host([has-scrolling-region]) > #contentContainer {
+      @apply(--layout-fit);
+      overflow-y: auto;
+      -webkit-overflow-scrolling: touch;
+    }
+
+    #contentContainer {
+      position: relative;
+      z-index: 1;
+      padding-top: 1px;
```

Comment:
That fixes this issue: http://jsbin.com/sodecuvubi/edit?html,output
because I'm using `margin-top` to move `#contentContainer` if the header has the `fixed`  attribute.